dashboard/app/models/user.rb

```diff
@@ -1981,6 +1981,15 @@ def undestroy
     restore(recursive: true, recovery_window: 5.minutes)
   end
 
+  def depended_upon_for_login?
+    # Teacher is depended upon for login if student does not have a personal login
+    # and student has no other teachers.
+    students.any? do |student|
+      student.can_create_personal_login? &&
+        student.sections_as_student.map(&:teacher).uniq.one?
```

Comment:
if there is a more efficient way to do this, please let me know! there used to be a [uniq_by](https://apidock.com/rails/Array/uniq_by) method on arrays, but it looks like that's been deprecated.

Reply:
Not more efficient, but you can say
```rb
student.teachers.uniq.one?
```
thanks to [this association](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user.rb#L415).  It's a little funny that `uniq` isn't implied in `student.teachers` but I just tried it out and it's not.

It might be _slightly_ more efficient to write
```rb
student.teachers.all? {|teacher| teacher == self}
```
which could sometimes early-out when a different teacher is found, where I think `uniq` has to check every element of the list.  But the set of teachers/sections for any particular student is so unlikely to hit a size where this matters, that I'd do whatever reads best to you.

Reply:
thank you! i'll go with `uniq.one?` as i think it reads a little more clearly